docs/management/persistence.md

```diff
@@ -259,26 +259,28 @@ and starts appending new data into the new file.
 
 ### How I can switch to AOF, if I'm currently using dump.rdb snapshots?
 
-There is a different procedure to do this in version 2.0 and later versions, as you
-can guess it's simpler since Redis 2.2 and does not require a restart at all.
+If you want to enable AOF in a server that is currently using RDB snapshots, you need to convert the data by enabling AOF via CONFIG command on the live server first.
+
+**IMPORTANT:** not following this procedure (e.g. just changing the config and restarting the server) can result in data loss!
 
 **Redis >= 2.2**
 
 * Make a backup of your latest dump.rdb file.
 * Transfer this backup to a safe place.
-* Issue the following two commands:
-* `redis-cli config set appendonly yes`
-* `redis-cli config set save ""`
-* Make sure your database contains the same number of keys it contained.
+* Enable AOF: `redis-cli config set appendonly yes`
+* Optionally disable RDB: `redis-cli config set save ""`
+* Make sure your database contains the same number of keys it contained previously.
```

Comment:
IMHO that line is excessive (and silly)

Reply:
I simply kept that line, which was already part of the docs before.

In general, a quick check that your data is still fine isn't totally wrong. Maybe we could word it like this?

    Make sure your database still contains all data, e.g. `redis-cli keys '*'`

Reply:
but it's silly since enabling AOF can't (technically) affect your data in redis.
maybe instead we can add a line that suggests to wait till that initial AOFRW process completes?
i.e. if you'll try to restart redis before that happens, it'll refuse to go down.

Reply:
sadly, i don't see that we exposed the status of `server.aof_state` in INFO, so we may need to suggest to look at some combination of `aof_rewrite_in_progress`, `aof_rewrite_scheduled`, and maybe `aof_last_bgrewrite_status`.
WDYT?

Reply:
> but it's silly since enabling AOF can't (technically) affect your data in redis.

That's what I thought, and then all data was gone after restarting Redis (and it took 4 hours for someone to notice) 😉 (I know what you mean though, from a technical point of view.)

> sadly, i don't see that we exposed the status of `server.aof_state` in INFO, so we may need to suggest to look at some combination of `aof_rewrite_in_progress`, `aof_rewrite_scheduled`, and maybe `aof_last_bgrewrite_status`. WDYT?

These are all keys that are listed when querying `INFO persistence`, correct? What would be the correct combination of keys? (I don't know enough about Redis internals to propose some wording.)

Reply:
> > but it's silly since enabling AOF can't (technically) affect your data in redis.
> 
> That's what I thought, and then all data was gone after restarting Redis (and it took 4 hours for someone to notice) 😉 (I know what you mean though, from a technical point of view.)

but that line tells you to check the data before restart, when all you did is change a configuration that doesn't affect the data in any way (redis doesn't touch it), so it's silly.
i.e. it does affect the guarantees, and how that data is persisted, but that's not what that line suggests to check.
IMHO we should delete it.

> > sadly, i don't see that we exposed the status of `server.aof_state` in INFO, so we may need to suggest to look at some combination of `aof_rewrite_in_progress`, `aof_rewrite_scheduled`, and maybe `aof_last_bgrewrite_status`. WDYT?
> 
> These are all keys that are listed when querying `INFO persistence`, correct? What would be the correct combination of keys? (I don't know enough about Redis internals to propose some wording.)

how about:
```
You may want to wait for AOF rewrite to finish persisting the data. You can do that by watching `INFO persistence`, waiting for `aof_rewrite_in_progress` and `aof_rewrite_scheduled` to be `0`, and validate that `aof_last_bgrewrite_status` is `ok`.
```

Reply:
This (check that the database contains the same number of keys it contained previously) makes sense to do after restarting. That's where things can go wrong.

I think it is better to move this bullet to a step after all other steps "Next time you restart the server, check that..."

I think we shall also add Oran's suggestion. It is important especially if you want to restart redis just after changing the config, when AOF rewrite is ongoing. I think we should also add "Before restarting, " to Oran's suggestion, so:

> Before restarting, you may want to wait for AOF rewrite to finish persisting the data. You can do that by watching `INFO persistence`, waiting for `aof_rewrite_in_progress` and `aof_rewrite_scheduled` to be `0`, and validate that `aof_last_bgrewrite_status` is `ok`.